### ISSUE TITLE
`controlledComputed` edge case when using "reactive"-like objects as values (eg. those returned by `toRefs`)

### ISSUE THREAD

**Issue by IlCallo** [vueuse/vueuse#289]
Really a niche edge-case, but maybe worth mentioning on the docs that `controlledComputed` is based on `ref`, and as such it cannot be used in scenarios like https://github.com/vuejs/composition-api/issues/629

Don't think it's worth to make it configurable, since it's so simple to just recreate the logic using `shallowRef`

**Comment by antfu** [comment]
What's your expected usage? Not sure if I got you correctly.

**Comment by IlCallo** [comment]
I tried to use `controlledComputed` to return an object containing refs as properties (using `toRefs`) every time a given condition is met. This won't work as you internally assign the generated value to a `ref`, which will automatically unref everything inside.

```ts
const source = ref('my-condition')
const fields = reactive({
  input: '',
  anotherInput: ''
})

const fieldRefs = controlledComputed(
  () => source.value,
  () => toRefs(fields)
)

console.log(fieldRefs.value)
// Will return
// { input: '', anotherInput: ''}
// I initially expected it to return
// { input: RefImpl, anotherInput: RefImpl}
```

Using a `shallowRef` internally should avoid the problem, but I don't know if there are any side effects with this change, so just documenting it would be cool

**Comment by antfu** [comment]
Thanks for the explanation, that's clear to me! I think we should use `shallowRef` as it would match with the behavior of `computed`. Would you mind creating a PR for this? Thanks!

**Comment by antfu** [comment]
Fixed in d1c7d4c24dec68cb1adb9a434502b001265b901f, thanks!

**Comment by IlCallo** [comment]
Thanks!
